compact.go

```diff
@@ -757,7 +757,13 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 		}
 
 		for i, chk := range chks {
-			if chk.MinTime < meta.MinTime || chk.MaxTime > meta.MaxTime {
+			// Re-encode head chunks that are still open(being written to) or
+			// outside the compacted MaxTime range.
+			// The chunk.Bytes() method is not safe for these chunks hence the re-encoding.
+			// This happens when snapshotting the head block.
+			if _, isHeadChunk := chk.Chunk.(*safeChunk); isHeadChunk && (chk.MaxTime > meta.MaxTime || chk.MaxTime == math.MaxInt64) {
```

Comment:
I think you don't need == math.MaxInt because MaxInt will be definitely bigger than `meta.MaxtTime` (:
```suggestion
			if _, isHeadChunk := chk.Chunk.(*safeChunk); isHeadChunk && (chk.MaxTime > meta.MaxTime) {
```